torch/_torch_docs.py

```diff
@@ -2906,6 +2904,8 @@ def merge_dicts(*dicts):
 
 Args:
     n (int): the number of rows
+
+Keyword arguments:
     m (int, optional): the number of columns with default being :attr:`n`
```

Comment:
The convention is that "Keyword arguments" are the keyword-only arguments. See torch.sub, for example:

https://pytorch.org/docs/master/generated/torch.sub.html?highlight=sub#torch.sub

Unrelated to this PR, of course, but I wish NumPy and PyTorch had been much more aggressive about making arguments keyword-only.

Reply:
Should we rename "Keyword arguments" then? The current name is plain wrong if it's keyword-only.

> Unrelated to this PR, of course, but I wish NumPy and PyTorch had been much more aggressive about making arguments keyword-only.

working on that: https://data-apis.github.io/array-api/latest/API_specification/function_and_method_signatures.html :)

Reply:
(that rename would be my second option), as I commented, separating out keyword-only just doesn't make much sense. Keywords like `dim` and `dtype` are _keywords_, and should preferably be used as such rather than as positional arguments.

Reply:
Okay, the `Keyword Arguments` name is coming from Napoleon: https://github.com/sphinx-contrib/napoleon/blob/83bf1963096490dd666f93ef5a9ed1cb229fc3ec/sphinxcontrib/napoleon/docstring.py#L673. Nowhere does Napoleon make clear whether that's meant as keyworl-only yes or no. The other odd thing is that Napoleon implements two docstring styles (Google or NumPy), and neither has `Keyword Arguments` as a separate section. Google just has `Args` and NumPy only `Parameters`.

Napoleon introduced it in [this commit](https://github.com/sphinx-contrib/napoleon/commit/43423d4834ce), which points to the Khan Academy style guide. But that style guide now follows Google style: https://github.com/Khan/style-guides/blob/master/style/python.md#docstrings.

tl;dr best to get rid of `Keyword Arguments` completely.

Reply:
Filed an issue for it at https://github.com/sphinx-contrib/napoleon/issues/29

Reply:
Want to also file an issue here? We could remove that section ourselves, too.

Do you really think it's not useful to separate kwarg-only arguments, though?

Reply:
> Do you really think it's not useful to separate kwarg-only arguments, though?

It confused me, and it serves no real purpose as far as I can tell - signatures contain the same info, plus non-only-named-params should typically also be used as keywords. And it's unusual, I have never seen it elsewhere.

On the other hand, I don't feel strongly enough about it to want to make changes that touch the whole code base at once.

It would be easier if we could rename the section to `Keyword-only Arguments` for clarity, but unfortunately that's difficult.

Maybe let's see what the Napoleon folks say first.